vm/vmstore/vm_store.go

```diff
@@ -157,14 +162,28 @@ func (v *VMContext) GetAccountMeta(address types.Address) (*types.AccountMeta, e
 }
 
 func (v *VMContext) SaveStorage(txns ...db.StoreTxn) error {
+	txn := v.Ledger.Store.NewTransaction(true)
+
 	storage := v.Cache.storage
 	for k, val := range storage {
-		err := v.set([]byte(k), val, txns...)
+		err := v.set([]byte(k), val, txn)
+		if err != nil {
+			v.logger.Error(err)
+			return err
+		}
+	}
+
+	delKeys := v.Cache.delete
+	for k, _ := range delKeys {
+		err := v.remove([]byte(k), txn)
 		if err != nil {
 			v.logger.Error(err)
 			return err
 		}
 	}
+
+	txn.Commit()
+	txn.Discard()
```

Comment:
we should not provider interface to remove data from contract, it is VERY dangerous